lib/vssh/libssh.c

```diff
@@ -1159,13 +1159,23 @@ static CURLcode myssh_statemach_act(struct Curl_easy *data, bool *block)
         break;
       }
       else if(statvfs) {
+        #ifdef _MSC_VER
+        #define LIBSSH_VFS_SIZE_MASK "llu"
+        #else
+        #define LIBSSH_VFS_SIZE_MASK PRIu64
+        #endif
```

Comment:
It's a C99 thing:
https://en.cppreference.com/w/c/types/integer

Reply:
In `libssh.h` v0.10.4 there is this, though it has changed since and [now](https://gitlab.com/libssh/libssh-mirror/-/blob/2df232463802203e2d675d4f99c65776a6f6b3c4/include/libssh/libssh.h#L54) includes `inttypes.h`, making it safe (probably) to use `PRIu64` everywhere:
```c
#ifdef _MSC_VER
  /* Visual Studio hasn't inttypes.h so it doesn't know uint32_t */
  typedef int int32_t;
  typedef unsigned int uint32_t;
  typedef unsigned short uint16_t;
  typedef unsigned char uint8_t;
  typedef unsigned long long uint64_t;
  typedef int mode_t;
#else /* _MSC_VER */
  #include <unistd.h>
  #include <inttypes.h>
  #include <sys/types.h>
#endif /* _MSC_VER */
```

Though for `_MSC_VER` I just left the value unchanged, as I didn't want to disturb what we had there (and cannot test it. CI also doesn't test it.).

Reply:
_msc_ver anything that can build curl will have __int64 (and unsigned __int64) and `%I64d` (and `%I64u`). `%llu` is not in older versions of the crt 

https://github.com/curl/curl/blob/d755a5f7c009dd63a61b2c745180d8ba937cbfeb/include/curl/system.h#L372-L385

Reply:
Meaning this mask was broken with MSVC all along?

Reply:
> Meaning this mask was broken with (old) MSVC all along?

Yes. It will compile without warning in older versions of visual studio but its output is the same as %lu